src/components/CodeBlock.vue

```diff
@@ -1,36 +1,99 @@
 <template>
-  <Prism
+  <PrismEditor
+    v-model="code"
     class="code-example"
-    :class="{ 'language-none': language === 'none' ? 'language-none' : '' }"
-    :language="language !== 'none' ? language : 'javascript'"
+    :class="classNames"
+    :highlight="highlighter"
+    :line-numbers="lineNumbers"
+    :readonly="!editable"
   >
+    <!--
+      @slot Codeblock content
+    -->
     <slot />
```

Comment:
Just out of interest, does this mean you're binding the `code` data property to the PrismEditor instance with `v-model` while also passing it as a slot?

Reply:
It's to allow for us to still use the slot like the previous integrations.

Reply:
I think this is a good use-case for a functional component. You're just returning a PrismEditor component, and you don't actually need to put the slot in the template - you're just accessing it's content with code. It might address the issues you were having with state updating too.